### ISSUE TITLE
Bug Report

### ISSUE THREAD

**Issue by rfrost77** [realm/realm-java#2402]
We launched app a week ago, and we got some issues that are suspected to be fired inside realm.
I want share them and hope it solved or get advice for solution.

1. http://crashes.to/s/6f786c3c97c
From BaseActivity, it is ours and line 130 of BaseActivity is initialization of realm.
```
if (mRealm == null) mRealm = Realm.getDefaultInstance();
```

2. http://crashes.to/s/24c6157fa0e
It has no our code. I really don't know why it happen.

3. http://crashes.to/s/d49020c314d
I removed findAllAsync() and findAllSortedAsync() because of BadVersion exception, but it happen again in findAllSorted(). I guess if I give multi parameter for sort, it sometimes fires. I want you to look at it too.

4. http://crashes.to/s/7aa35297440
It also happen line 130 of BaseActivity, initialization of realm (above code). I event don't know why array exception is here.

Thanks.

**Comment by kneth** [comment]
Congratulation on the launch!

Without the actual source code - or snippets - it can be hard to give you optimal advices but I'll do my best.

1. The error is that the format of the Realm file is invalid. If you use encryption, changing the encryption will lead to this error.

2. My best guess is that your device ran out of disk space and one of the aux. files used by Realm was truncated. We have just merged #2397 which inlines these aux. files.

3. This should be solved by #2397.

4. This is an interesting one. Realm stores some metadata (in an array-like structure), and it seems that you get it when opening a Realm file (getting an instances). Do you create a `RealmConfiguration` and set the default configuration to it? If you do, can you share it?

**Comment by dramancompany** [comment]
@kneth  1. I did not use any encryption settings myself. Then, how can I solve it?

2, 3 seems what I can do is just wait 0.88 ...

4. I attach my code of realm initialization.
```
public static RealmConfiguration.Builder mRealmBuilder = new RealmConfiguration.Builder(BaseApplication.get())
            .name(Realm.DEFAULT_REALM_NAME)
            .schemaVersion(SCHEMA_VERSION)
            .migration(new MigrationRealmUtil());

    public static void initialize() {
        if (AppUtil.isDebugMode()) mRealmBuilder.deleteRealmIfMigrationNeeded();
        Realm.setDefaultConfiguration(mRealmBuilder.build());
    }
```
Can you analyze the reason of the bug through above code?

Thanks.

**Comment by rfrost77** [comment]
@kneth  1. I did not use any encryption settings myself. Then, how can I solve it?

2, 3 seems what I can do is just wait 0.88 ...

4. I attach my code of realm initialization.
```
public static RealmConfiguration.Builder mRealmBuilder = new RealmConfiguration.Builder(BaseApplication.get())
            .name(Realm.DEFAULT_REALM_NAME)
            .schemaVersion(SCHEMA_VERSION)
            .migration(new MigrationRealmUtil());

    public static void initialize() {
        if (AppUtil.isDebugMode()) mRealmBuilder.deleteRealmIfMigrationNeeded();
        Realm.setDefaultConfiguration(mRealmBuilder.build());
    }
```
Can you analyze the reason of the bug through above code?

Thanks.

**Comment by bmunkholm** [comment]
@rfrost77 Could I kindly request you to create separate issues for the outstanding ones with as much details as possible so we can better track and resolve the issues? Thanks!

**Comment by beeender** [comment]
@rfrost77 
> MigrationRealmUtil still does not have any code inside. 

Then how do you handle `RealmMigrationNeededException`? What is the current`SCHEMA_VERSION` now?

**Comment by beeender** [comment]
@rfrost77 Any updates on this?

**Comment by rfrost77** [comment]
@beeender current schema version is 2 and I used deleteRealmIfMigrationNeeded() cause I did not need migrate DB.

**Comment by beeender** [comment]
The No.3 has been reported in #2522 . We are waiting for device to reproduce it.

**Comment by bmunkholm** [comment]
@rfrost: We can also split the issue into multiple, but you won't get notifications automatically then. It would be really great if you would split up the issues in separate github issues. Thanks!

**Comment by beeender** [comment]
1. Dup to #2524 
2. #2567 created for tracking the issue
3. Should be fixed by #2450 
4. Dup to #2522 

@rfrost77 For easier tracking, i created/duplicated those issues into individual ones. See above. And I am closing this one since it is not easy to have discussion with many issues in one thread. Let's continue discussion in corresponding individual issues.
